hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/scheduledexecutor/ScheduledExecutorTaskCancelFromAddressMessageTask.java

```diff
@@ -0,0 +1,89 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.impl.protocol.task.scheduledexecutor;
+
+import com.hazelcast.client.impl.protocol.ClientMessage;
+import com.hazelcast.client.impl.protocol.codec.ScheduledExecutorCancelFromAddressCodec;
+import com.hazelcast.client.impl.protocol.task.AbstractAddressMessageTask;
+import com.hazelcast.instance.Node;
+import com.hazelcast.nio.Address;
+import com.hazelcast.nio.Connection;
+import com.hazelcast.scheduledexecutor.ScheduledTaskHandler;
+import com.hazelcast.scheduledexecutor.impl.DistributedScheduledExecutorService;
+import com.hazelcast.scheduledexecutor.impl.ScheduledTaskHandlerImpl;
+import com.hazelcast.scheduledexecutor.impl.operations.CancelTaskOperation;
+import com.hazelcast.security.permission.ActionConstants;
+import com.hazelcast.security.permission.ScheduledExecutorPermission;
+import com.hazelcast.spi.Operation;
+
+import java.security.Permission;
+
+public class ScheduledExecutorTaskCancelFromAddressMessageTask
+        extends AbstractAddressMessageTask<ScheduledExecutorCancelFromAddressCodec.RequestParameters> {
+
+    public ScheduledExecutorTaskCancelFromAddressMessageTask(ClientMessage clientMessage, Node node, Connection connection) {
+        super(clientMessage, node, connection);
+    }
+
+    @Override
+    protected Operation prepareOperation() {
+        ScheduledTaskHandler handler = ScheduledTaskHandlerImpl.of(parameters.address,
+                parameters.schedulerName,
+                parameters.taskName);
+        return new CancelTaskOperation(handler, parameters.mayInterruptIfRunning);
+    }
+
+    @Override
+    protected Address getAddress() {
+        return parameters.address;
+    }
+
+    @Override
+    protected ScheduledExecutorCancelFromAddressCodec.RequestParameters decodeClientMessage(ClientMessage clientMessage) {
+        return ScheduledExecutorCancelFromAddressCodec.decodeRequest(clientMessage);
+    }
+
+    @Override
+    protected ClientMessage encodeResponse(Object response) {
+        return ScheduledExecutorCancelFromAddressCodec.encodeResponse((Boolean) response);
+    }
+
+    @Override
+    public String getServiceName() {
+        return DistributedScheduledExecutorService.SERVICE_NAME;
+    }
+
+    @Override
+    public Permission getRequiredPermission() {
+        return new ScheduledExecutorPermission(parameters.schedulerName, ActionConstants.ACTION_MODIFY);
+    }
+
+    @Override
+    public String getDistributedObjectName() {
+        return parameters.schedulerName;
+    }
+
+    @Override
+    public String getMethodName() {
+        return "cancel";
+    }
+
+    @Override
+    public Object[] getParameters() {
+        return new Object[] { parameters.mayInterruptIfRunning };
```

Comment:
ok, but these parameters are used for the security interceptor, and even though one can get the name of the scheduler (distributedObject), the action name (eg. cancel) and the mayInterrupt flag, they can't see which task this is targeted to. So **unless I am missing sth**, we are missing the taskName here at the very least ?
Also, [this](https://github.com/hazelcast/hazelcast/pull/9862/commits/1a3cf3bf7ba1201418a300865f3b178dd754a4c4#diff-2f3f5d0869f5bd6f5adcd16b6339bf9cR93) needs to adhere to the same pattern.